src/global_sequences_viewer.cpp

```diff
@@ -0,0 +1,169 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "global_sequences_viewer.hpp"
+
+#include <memory>
+#include <utility>
+
+#include "logging.hpp"
+#include "model.hpp"
+#include "modelversion.hpp"
+#include "statefulmodelinstance.hpp"
+#include "status.hpp"
+
+namespace ovms {
+
+static bool sequenceWatcherStarted = false;
+
+ovms::Status GlobalSequencesViewer::addVersions(std::shared_ptr<ovms::Model>& model,
+    std::shared_ptr<model_versions_t> versionsToAdd,
+    std::shared_ptr<model_versions_t> versionsFailed) {
+    for (const auto version : *versionsToAdd) {
+        if (std::count(versionsFailed->begin(), versionsFailed->end(), version))
+            continue;
+        auto modelInstance = model->getModelInstanceByVersion(version);
+        auto stetefulModelInstance = std::static_pointer_cast<StatefulModelInstance>(modelInstance);
+        std::string managerId = model->getName() + std::to_string(version);
+
+        auto status = registerManager(managerId, stetefulModelInstance->getSequenceManager().get());
+        if (status.getCode() != ovms::StatusCode::OK)
+            return status;
+    }
+
+    updateThreadInterval();
+    return ovms::StatusCode::OK;
+}
+
+ovms::Status GlobalSequencesViewer::retireVersions(std::shared_ptr<ovms::Model>& model, std::shared_ptr<model_versions_t> versionsToRetire) {
+    for (const auto version : *versionsToRetire) {
+        std::string managerId = model->getName() + std::to_string(version);
```

Comment:
I think we need separator here. What if someone has model with name "model" with version 12 and "model1" with version 2? They will be translated to the same key and will result in conflict at some point.